### PR TITLE
Error Codes for NodeJS compatibility added

### PR DESCRIPTION
**Pull request by smittals2**
### Description of changes: 
Added in some error codes from OpenSSL 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and the ISC license.

### CODE REVIEW

File: include/openssl/err.h [hunk 1]
```diff
@@ -339,7 +339,17 @@ enum {
...
+  ERR_LIB_PKCS12,
+  ERR_LIB_DSO=37,
+  ERR_LIB_OSSL_STORE=44,
+  ERR_LIB_FIPS,
+  ERR_LIB_CMS,
+  ERR_LIB_TS,
+  ERR_LIB_CT=50,
+  ERR_LIB_ASYNC,
+  ERR_LIB_KDF,
+  ERR_LIB_SM2
```
Comment (WillChilds-Klein) [inline include/openssl/err.h:352]: in light of [this commit](https://github.com/aws/aws-lc/commit/6bdd10042ea5e9015bb1ecf740b76a29ea7c0302), you'll need to convert these into macros. do we know why there are explicit values assigned to some of these enum entries?

**Review by WillChilds-Klein — CHANGES_REQUESTED** [review]

File: include/openssl/err.h (hunk 1 above)
Comment (smittals2) [reply include/openssl/err.h:352]: Ah, I'll make that change. 

There were many other error codes in OpenSSL that I have not added here, only those required by NodeJS. By providing explicit values I've tried to maintain that same values for those error codes across libraries. However, the values shouldn't matter so I can also make them numerically ascending if we prefer that

Comment (smittals2) [reply include/openssl/err.h:352]: https://github.com/openssl/openssl/blob/5677992679b38950c6a0c3775fd57378e1879ba5/include/openssl/err.h.in#L113

File: include/openssl/dh.h [hunk 2]
```diff
@@ -385,4 +385,8 @@ BSSL_NAMESPACE_END
 #define DH_R_INVALID_NID 106
 #define DH_R_INVALID_PARAMETERS 107
 
+// error code for nodejs compatibility
```
Comment (WillChilds-Klein) [inline include/openssl/dh.h:388]: nit: we might want to remove this comment. other projects we integrate with may depend on this as well, nodejs just happens to be the first one.

**Review by WillChilds-Klein — APPROVED** [review]

File: include/openssl/dh.h (hunk 2 above)
Comment (andrewhop) [inline include/openssl/dh.h:388]:
```suggestion
```

File: include/openssl/err.h [hunk 3]
```diff
@@ -339,6 +339,16 @@ OPENSSL_EXPORT int ERR_get_next_error_library(void);
...
+#define ERR_LIB_PKCS12          35
+#define ERR_LIB_DSO             36
+#define ERR_LIB_OSSL_STORE      37
+#define ERR_LIB_FIPS            38
+#define ERR_LIB_CMS             39
+#define ERR_LIB_TS              40
+#define ERR_LIB_CT              41
+#define ERR_LIB_ASYNC           42
+#define ERR_LIB_KDF             43
+#define ERR_LIB_SM2             44
```
Comment (andrewhop) [inline include/openssl/err.h:351]: Hmmmmmmm, does NodeJS not handle `OPENSSL_NO_SM2` correctly?

Comment (smittals2) [reply include/openssl/err.h:351]: Not sure but they define mappings to these specific ERR_LIB_* codes, so we need to define them as well.

Comment (andrewhop) [reply include/openssl/err.h:351]: Where do you see that? A naive search only finds the OpenSSL dependency code and no real usage https://github.com/search?q=repo%3Anodejs%2Fnode%20ERR_LIB_SM2&type=code

Comment (smittals2) [reply include/openssl/err.h:351]: They create a mapping to those err codes here:
https://github.com/nodejs/node/blob/1abff073921bcb0631602032aef0135bccfaee0d/src/crypto/crypto_util.cc#L501

**Review by WillChilds-Klein — APPROVED** [review]

**Review by andrewhop — APPROVED** [review]

**Pull request merged**